sklearn/metrics/_plot/tests/test_plot_precision_recall.py

```diff
@@ -190,3 +192,50 @@ def test_default_labels(pyplot, average_precision, estimator_name,
                                   estimator_name=estimator_name)
     disp.plot()
     assert disp.line_.get_label() == expected_label
+
+
+@pytest.mark.parametrize(
+    "response_method", ["predict_proba", "decision_function"]
+)
+def test_plot_precision_recall_pos_label(pyplot, response_method):
+    # check that we can provide the positive label and display the proper
+    # statistics
+    X, y = load_breast_cancer(return_X_y=True)
+    # create an highly imbalanced
+    idx_positive = np.flatnonzero(y == 1)
+    idx_negative = np.flatnonzero(y == 0)
+    idx_selected = np.hstack([idx_negative, idx_positive[:25]])
+    X, y = X[idx_selected], y[idx_selected]
+    X, y = shuffle(X, y, random_state=42)
+    # only use 2 features to make the problem even harder
+    X = X[:, :2]
+    y = np.array(
+        ["cancer" if c == 1 else "not cancer" for c in y], dtype=object
+    )
+    X_train, X_test, y_train, y_test = train_test_split(
+        X, y, stratify=y, random_state=0,
+    )
+
+    classifier = LogisticRegression()
+    classifier.fit(X_train, y_train)
+
+    # sanity check to be sure the positive class is classes_[0] and that we
+    # are betrayed by the class imbalance
+    assert classifier.classes_.tolist() == ["cancer", "not cancer"]
+
+    disp = plot_precision_recall_curve(
+        classifier, X_test, y_test, pos_label="cancer",
+        response_method=response_method
+    )
+    # we should obtain the statistics of the "cancer" class
+    avg_prec_limit = 0.65
+    assert disp.average_precision < avg_prec_limit
```

Comment:
I thought of suggesting something similar to what you did in #17594

```py
y_pred = y_pred[:, 0] if y_pred.ndim == 2 else -y_pred
```

And then compute the average precision. Do you have a preference?

Reply:
if the average precision is bugged then we will be in trouble (eg as with the roc_auc_score)
At least here I am comparing the area under the curve which should be close to the average precision.
Having hard coded is at least one way to detect if something goes side way